lib/menu.rb

```diff
@@ -0,0 +1,57 @@
+require 'csv'
+require_relative 'dish'
+
+class Menu
+
+  OUTPUT_WIDTH = 60
+
+  attr_reader :dishes
+
+  def initialize(filename = 'menu.csv')
+    @filename = filename
+    load_menu
+  end
+
+  def load_menu(filename = @filename)
+    file_not_exist?(filename)
+    load_data_into_hash(filename)
+  end
+
+  def print_menu(output_width = OUTPUT_WIDTH)
+    print_header(output_width)
+    print_body(output_width)
+  end
+
+  private
+
+  def load_data_into_hash(filename)
+    @dishes = Hash.new
+    CSV.foreach(filename) do |row|
+      dish = Dish.new
+      dish.name, dish.ingredients, dish.price, dish.type  = row[0], row[1], row[2].to_i, row[3]
+      @dishes.store(@dishes.length+1,dish)
```

Comment:
Tried to do it with an array, but I found it inconvenient to iterate through it. Why using hashes is worse than using arrays?